File: Ros_Study/my_message/src/listener.cpp
#include "ros/ros.h"
#include "my_message/chat_msg.h"

void callback(const my_message::chat_msg::ConstPtr& msg)
{
    ROS_INFO("I heard [%s] said '%s'", msg -> name.c_str(), msg -> chat.c_str());
}



int main(int argc, char **argv)
{
    ros::init(argc, argv, "listener_cpp");
    ros::NodeHandle nh;
    ros::Subscriber sub = nh.subscribe("chatter", 10, callback);

    ros::spin();

    return 0;
}

File: Ros_Study/my_message/scripts/listener.py
#!/usr/bin/env python

# -*- coding: utf-8 -*-
# @Author: IBNBlank
# @Date:   2019-01-28 22:50:17
# @Last Modified by:   IBNBlank
# @Last Modified time: 2019-01-29 21:54:21

import rospy
from my_message.msg import chat_msg


def callback(data):
    rospy.loginfo("I heard [{0}] said '{1}'".format(data.name, data.chat))



if __name__ == '__main__':
    rospy.init_node("listener_py", anonymous=True)
    sub = rospy.Subscriber("chatter", chat_msg, callback, queue_size=10)

    rospy.spin()
